### PR TITLE
Zy/2 initialise pod

### PR DESCRIPTION
**Pull request by anushkavidanage**
# Pull Request Details
Merging the branch with dev to have the new package name `solidpod`

## What issue does this PR address
- resolving all the conflicts when merging from dev

- Link to associated issue: https://github.com/anusii/solidpod/issues/2

## Checklist

[Use the check-list below to ensure your branch is ready for PR]

- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [ ] No duplicated content
- [ ] No lint errors (`make prep` or `flutter analyze lib`
                       python: use pre-commit ruff hook to check on commit)
- [x] No lint check errors related to your changes
- [ ] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome
  - [ ] iOS
  - [x] Linux
  - [ ] MacOS
  - [ ] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number
- [ ] Merge PR into dev

### CODE REVIEW

**Review by zheyxu — APPROVED** [review]

File: lib/src/solid/authenticate.dart [hunk 1]
```diff
@@ -30,9 +30,8 @@
 
 library;
 
-import 'package:flutter/material.dart';
-
 import 'package:fast_rsa/fast_rsa.dart';
+import 'package:flutter/material.dart';
```
Comment (gjwgit) [inline lib/src/solid/authenticate.dart:34]: This should be separate line and first.

File: lib/src/solid/login.dart [hunk 2]
```diff
@@ -31,14 +31,16 @@
 library;
 
 import 'package:flutter/material.dart';
```
Comment (gjwgit) [inline lib/src/solid/login.dart:33]: This is all correct re ordering imports - please follow this style.

File: lib/src/solid/login.dart [hunk 3]
```diff
@@ -31,14 +31,16 @@
...
-
 import 'package:package_info_plus/package_info_plus.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 import 'package:solidpod/src/solid/authenticate.dart';
```
Comment (gjwgit) [inline lib/src/solid/login.dart:37]: Except these should be alphabetic.

**Review by gjwgit — CHANGES_REQUESTED** [review]
Please don't bring this into dev until it has been re-engineered. 

For example, initial_setup_desktop needs to be cleaned up and made a lot smaller.

`make prep` needs a lot of work. I did a `make fix` to autofix what can be done.

Great that it is functional though and we can test the initialise in this branch to at least unblock other tasks.

**Comment by anushkavidanage** [comment]
@gjwgit I have now completely restructured the files for this functionality. Please have a look and let me know any changes are required.

**Comment by gjwgit** [comment]
I am seeing this dependency issue - I've updated flutter to latest. Perhaps that caused it?

```
23 Feb 08:27:53 gjw@kadesh ~/projects/keypod$ ls
total 160K
4.0K analysis_options.yaml  4.0K build/        4.0K guidelines/   36K LICENSE       4.0K Makefile         0 solidpod@  4.0K windows/
4.0K android/               4.0K CHANGELOG.md  4.0K ios/         4.0K license.dart   36K pubspec.lock  4.0K support/
4.0K assets/                4.0K coverage/     4.0K keypod.iml   4.0K linux/        4.0K pubspec.yaml  4.0K test/
4.0K backup/                4.0K doc/          4.0K lib/         4.0K macos/        4.0K README.md     4.0K web/
23 Feb 08:29:26 gjw@kadesh ~/projects/keypod$ cd solidpod
/home/gjw/projects/keypod/solidpod
23 Feb 08:29:38 gjw@kadesh ~keypod/solidpod$ git pull
Already up to date.
23 Feb 08:29:43 gjw@kadesh ~keypod/solidpod$ git status
On branch zy/2_initialise_pod
Your branch is up to date with 'origin/zy/2_initialise_pod'.

nothing to commit, working tree clean
23 Feb 08:29:46 gjw@kadesh ~keypod/solidpod$ cd ..
23 Feb 08:29:52 gjw@kadesh ~/projects/keypod$ make linux
flutter run --device-id linux
Resolving dependencies... (1.0s)
Because solid_auth 0.1.13 depends on intl ^0.18.0 and no versions of solid_auth match >0.1.13 <0.2.0, solid_auth ^0.1.13 requires intl ^0.18.0.
And because every version of solidpod from path depends on intl ^0.19.0, solid_auth ^0.1.13 is incompatible with solidpod from path.
So, because keypod depends on solidpod from path which depends on solid_auth ^0.1.13, version solving failed.
make: *** [support/flutter.mk:92: linux] Error 1
23 Feb 08:30:26 gjw@kadesh ~/projects/keypod$ flutter --version
Flutter 3.19.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision abb292a07e (2 days ago) • 2024-02-20 14:35:05 -0800
Engine • revision 04817c99c9
Tools • Dart 3.3.0 • DevTools 2.31.1
```

**Comment by anushkavidanage** [comment]
Resolved this by lowering the `intl` version since not only `solid_auth`, but a few other packages also depend on `intl` 18.0.

**Review by gjwgit — APPROVED** [review]
Looking good. Thanks for that.

**Comment by gjwgit** [comment]
With @anushkavidanage to merge the PR.

**Pull request merged**